plugins/system/tour/tour.php

```diff
@@ -0,0 +1,189 @@
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function onBeforeRender()
+	{
+		// Run in backend
+		if ($this->app->isClient('administrator'))
+		{
+			/**
+			 * Booting of the Component to get the data in JSON Format
+			 */
+			$myTours = $this->app->bootComponent('com_guidedtours')->getMVCFactory()->createModel('Tours', 'Administrator', ['ignore_request' => true]);
+			$mySteps = $this->app->bootComponent('com_guidedtours')->getMVCFactory()->createModel('Steps', 'Administrator', ['ignore_request' => true]);
+
+			$theCurrentExtension = $this->app->input->get('option');
+			$myTours->setState('list.extensions', $theCurrentExtension);
+
+			$myTours->setState('filter.published', 1);
+
+			$steps = $mySteps->getItems();
+			$tours = $myTours->getItems();
+
+			$document = Factory::getDocument();
+
+			$newsteps = [];
+
+			foreach ($steps as $step)
+			{
+				if (!isset($newsteps[$step->tour_id]))
+				{
+					$newsteps[$step->tour_id] = [];
+				}
+
+				$newsteps[$step->tour_id][] = $step;
+			}
+
+			foreach ($tours as $tour)
+			{
+				$tour->steps = [];
+
+				if (isset($newsteps[$tour->id]))
+				{
+					$tour->steps = $newsteps[$tour->id];
+				}
+			}
+
+			$myTours = json_encode($tours);
+			$mySteps = json_encode($steps);
+			$document->addScriptOptions('myTours', $myTours);
+			$document->addScriptOptions('mySteps', $mySteps);
+
+			$toolbar = Toolbar::getInstance('toolbar');
+			$dropdown = $toolbar->dropdownButton()
+				->text('Take the Tour')
+				->toggleSplit(false)
+				->icon('fa fa-map-signs')
+				->buttonClass('btn btn-action');
+
+			$childBar = $dropdown->getChildToolbar();
+
+			foreach ($tours as $a)
+			{
+				$childBar->BasicButton('tour')
+					->text($a->title)
+					->attributes(['data-id' => $a->id])
+					->buttonClass('btn btn-primary');
+			}
+		}
+	}
+
+	/**
+	 * Listener for the `onBeforeCompileHead` event
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function onBeforeCompileHead()
+	{
+
+		if ($this->app->isClient('administrator'))
+		{
+			HTMLHelper::_(
+				'script',
+				Uri::root() . 'build/media_source/plg_system_tour/js/guide.js',
+				array('version' => 'auto', 'relative' => true)
+			);
+
+			HTMLHelper::_(
+				'script',
+				Uri::root() . 'build/media_source/plg_system_tour/js/shepherd.min.js',
+				array('version' => 'auto', 'relative' => true)
+			);
+			HTMLHelper::_(
+				'script',
+				Uri::root() . 'build/media_source/plg_system_tour/js/popper.min.js',
+				array('version' => 'auto', 'relative' => true)
+			);
+			HTMLHelper::_(
+				'stylesheet',
+				Uri::root() . 'build/media_source/plg_system_tour/css/shepherd.css',
+				array('version' => 'auto', 'relative' => true)
+			);
+			HTMLHelper::_(
+				'stylesheet',
+				Uri::root() . 'build/media_source/plg_system_tour/css/shepherd.min.css',
+				array('version' => 'auto', 'relative' => true)
+			);
+			HTMLHelper::_(
+				'stylesheet',
+				Uri::root() . 'build/media_source/plg_system_tour/css/guide.css',
```

Comment:
```suggestion
				Uri::root() . 'media/plg_system_tour/css/guide.css',
```

Reply:
No, shepherd should be loaded as preset and the tour.js as a relative file via the webasset manager

Reply:
So @bembelimen lines 155 to 188 or thereabouts are redundant and need to be moved to the webasset manager for loading?